firstvoices/backend/views/import_job_views.py

```diff
@@ -0,0 +1,73 @@
+from django.utils.translation import gettext as _
+from drf_spectacular.utils import OpenApiResponse, extend_schema, extend_schema_view
+from rest_framework import parsers
+from rest_framework.viewsets import ModelViewSet
+
+from backend.models.import_jobs import ImportJob
+from backend.views.base_views import FVPermissionViewSetMixin, SiteContentViewSetMixin
+
+from ..serializers.import_job_serializers import ImportJobSerializer
+from . import doc_strings
+from .api_doc_variables import id_parameter, site_slug_parameter
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description=_(
+            "A list of batch import jobs associated with the specified site."
+        ),
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_list,
+                response=ImportJobSerializer,
+            ),
+            403: OpenApiResponse(description=doc_strings.error_403_site_access_denied),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    retrieve=extend_schema(
+        description=_("Details about a specific batch import job."),
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_detail,
+                response=ImportJobSerializer,
+            ),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+    create=extend_schema(
+        description=_("Add a batch import job."),
+        responses={
+            201: OpenApiResponse(
+                description=doc_strings.success_201, response=ImportJobSerializer
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+)
+class ImportJobViewSet(SiteContentViewSetMixin, FVPermissionViewSetMixin, ModelViewSet):
+    serializer_class = ImportJobSerializer
+    http_method_names = ["get", "post", "delete"]
+    parser_classes = [
+        parsers.FormParser,
+        parsers.MultiPartParser,  # to support file uploads
+        parsers.JSONParser,
+    ]
+
+    def get_queryset(self):
+        site = self.get_validated_site()
+        return ImportJob.objects.filter(
+            site__slug=site[0].slug
```

Comment:
The double underscore in this filter will do the lookup by joining the site table. For a faster lookup, you can just do site=site[0] and it will check the site_id field in the ImportJob table directly